universal-application-tool-0.0.1/app/services/applicant/predicate/PredicateEvaluator.java

```diff
@@ -0,0 +1,49 @@
+package services.applicant.predicate;
+
+import services.applicant.ApplicantData;
+import services.applicant.exception.InvalidPredicateException;
+import services.program.predicate.LeafOperationExpressionNode;
+import services.program.predicate.PredicateExpressionNode;
+
+/** Evaluates complex predicates based on the given {@link ApplicantData}. */
+public class PredicateEvaluator {
+
+  private final ApplicantData applicantData;
+  private final JsonPathPredicateGenerator predicateGenerator;
+
+  public PredicateEvaluator(
+      ApplicantData applicantData, JsonPathPredicateGenerator predicateGenerator) {
+    this.applicantData = applicantData;
+    this.predicateGenerator = predicateGenerator;
+  }
+
+  /**
+   * Evaluate an expression tree rooted at the given {@link PredicateExpressionNode}. Will return
+   * true if and only if the entire tree evaluates to true based on the {@link ApplicantData} used
+   * to create this generator.
```

Comment:
(nit) I think "this generator" should be "this evaluator".

Reply:
Good catch! Fixed